var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -657,36 +658,35 @@ def __init__(self, spec, **kwargs):
         if '+float32' in spec:
             self.precision_option = 'SP'
 
-        # spec.architecture.platform is like `uname -s`, but lower-case
-        platform = spec.architecture.platform
+        # OpenFOAM chooses which wmake rules to use based on the output of
+        # `uname -s` and `uname -m`.  We can get the same information from
+        # Python's platform library.
 
-        # spec.architecture.target is like `uname -m`
-        target   = spec.architecture.target
+        system = platform.system().lower()
+        machine = platform.machine()
 
-        if platform == 'linux':
-            if target == 'i686':
+        self.arch = system
+        self.arch_option = '64'
+
+        if system == 'linux':
+            if machine == 'i686':
                 self.arch_option = '32'  # Force consistency
-            elif target == 'x86_64':
-                if self.arch_option == '64':
-                    platform += '64'
-            elif target == 'ia64':
-                platform += 'ia64'
-            elif target == 'armv7l':
-                platform += 'ARM7'
-            elif target == 'ppc64':
-                platform += 'PPC64'
-            elif target == 'ppc64le':
-                platform += 'PPC64le'
-        elif platform == 'darwin':
-            if target == 'x86_64':
-                platform += 'Intel'
+            elif machine == 'x86_64':
+                self.arch += '64'
```

Comment:
I wasn't sure what to do with `arch_option`.  It's set to the default `self.arch_option = '64'` at the beginning of `__init__()`, and isn't set to `'32'` anywhere except inside the check for `i686`.